notesapi/v1/views.py

```diff
@@ -109,31 +115,49 @@ class AnnotationSearchView(GenericAPIView):
 
             * updated: DateTime. When was the last time annotation was updated.
     """
+    params = {}
+    query_params = {}
+    search_with_usage_id = False
 
     def get(self, *args, **kwargs):  # pylint: disable=unused-argument
         """
         Search annotations in most appropriate storage
         """
+        self.query_params = {}
+        self.search_with_usage_id = False
+        self.params = self.request.query_params.dict()
+
+        usage_ids = self.request.query_params.getlist('usage_id')
+        if len(usage_ids) > 0:
+            self.search_with_usage_id = True
+            self.query_params['usage_id__in'] = usage_ids
+
+        if 'course_id' in self.params:
+            self.query_params['course_id'] = self.params['course_id']
+
         # search in DB when ES is not available or there is no need to bother it
-        if settings.ES_DISABLED or 'text' not in self.request.query_params.dict():
+        if settings.ES_DISABLED or 'text' not in self.params:
+            if 'user' in self.params:
+                self.query_params['user_id'] = self.params['user']
             return self.get_from_db(*args, **kwargs)
         else:
+            if 'user' in self.params:
+                self.query_params['user'] = self.params['user']
             return self.get_from_es(*args, **kwargs)
 
     def get_from_db(self, *args, **kwargs):  # pylint: disable=unused-argument
         """
         Search annotations in database.
         """
-        params = self.request.query_params.dict()
-        query = Note.objects.filter(
-            **{f: v for (f, v) in params.items() if f in ('course_id', 'usage_id')}
-        ).order_by('-updated')
+        query = Note.objects.filter(**self.query_params).order_by('-updated')
 
-        if 'user' in params:
-            query = query.filter(user_id=params['user'])
+        if 'text' in self.params:
+            query = query.filter(Q(text__icontains=self.params['text']) | Q(tags__icontains=self.params['text']))
 
-        if 'text' in params:
-            query = query.filter(Q(text__icontains=params['text']) | Q(tags__icontains=params['text']))
+        # Do not send paginated result if usage id based search.
```

Comment:
@mushtaqak What is the reason behind don't paginate if `usage_id` is in query params? Is this a safe assumption?